app/src/main/java/com/github/factotum_sdp/factotum/ui/roadbook/DRecordDetailsFragment.kt

```diff
@@ -74,8 +81,27 @@ class DRecordDetailsFragment : Fragment() {
             putBoolean("isSubFragment", true)
         }
 
+        var currentContact: Contact? = null
+        try {
+            currentContact = contactsViewModel.contacts.value?.first { c -> c.username == rec.clientID }
+        } catch (e: NoSuchElementException) {
+            // Handle the exception here
+            Log.e("Error", "Could not find contact with ID ${rec.clientID}")
```

Comment:
Super nice to think about logging a possibly main issue during the execution of the app.